src/Inviqa/Emarsys/Api/Response/ContactResponse.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Inviqa\Emarsys\Api\Response;
+
+class ContactResponse
+{
+    private $isSuccessful;
+
+    /**
+     * @throws \LogicException
+     */
+    public static function fromClientResponse(ClientResponse $response): self
+    {
+        $instance = new self();
+
+        $instance->isSuccessful = $response->isSuccessful();
+        $instance->validate($response);
```

Comment:
If the response is invalid, the `validate()` method will throw an exception, so the new instance will never be returned.
As a result `$instance->isSuccessful()` never will return false.

Reply:
The Exception is caught in the Client, and a new response created there.

Reply:
But that does not explain or solve the problem I mentioned.